clinton-hall/nzbToMedia
# Code cleanup core

##### DESCRIPTION

This PR focuses on code clean-up more of `core/*`.

While the majority of this PR is just code cleanup for things like PEP8 compliance, the important commits are the following:

##### IMPORTANT CHANGE LIST:
* 3acaf29 Update `six` to 1.10.0
* 382d108 Optimize imports
 * PEP8: Fix module level import not at top of file
 * Remove unused imports
 * Remove `simplejson`
 * Replace `mimetools` with `email`
* eb1ee8b `set` expected, but passing `list`
* Python 2/3 compatibility
 * cf1ae93 Use `six` to standardize imports
 * d4e5809 Use `print_function` to standardize printing
 * abf63d6 Use `six.iteritems` helper to standardizes `dict.iteritems`
 * ec2fc8d Use `six.text_type` to standardize `unicode` function
 * 5903538 Make `long = int` in Python 3
 * ec71e78 Convert `except ExceptionClass, Target:` to `except ExceptionClass as Target:`
 * 8434fd5 Fix relative imports
 * 22daf02 `raise` with no arguments can only be used in an `except` block
* df8c6bc Too broad exceptions: Use `.get()` with default value instead.

##### REMAINING CHANGE LIST:
* Add encoding declaration
* Convert to new-style class by inheriting from `object`
* PEP8: Fix formatting
 * Remove redundant backslash between brackets
 * Fix multiple statements on one line
 * Fix missing/excess whitespace
 * Fix comments not starting with a single `#` and a space
 * Convert tabs to spaces
 * Use triple-quoted docstring
* PEP8: Tests for membership should use `in`/`not in`
* PEP8: comparison to `None`, `True`, or `False` should use `is`/`is not`
* PEP8: `.has_key()` is deprecated, use `in`
* Streamline conditional statements
 * Non-zero `int` evaluate True.
* Code cleanup
 * Streamline variable assignment
 * Replace assignment with augmented assignment
 * Remove unused variables and redundant parentheses
* Rewrite dictionary creation as a dictionary literal
* Use `format()` instead of `%` for string formatting

Note: This PR already includes #1048
